test/Test/Time/Units.hs

```diff
@@ -32,8 +34,23 @@ spec_convertUnit =
         it "3 seconds is 3000000 microseconds" $
             convertUnit @MicroSecondUnit (sec 3) `shouldBe` 3000000
         it "3 microseconds is 3/1000000 seconds" $
-            convertUnit @SecondUnit (mcs 3) `shouldBe` Time (3 % 1000000)
+            convertUnit @SecondUnit (mcs 3) `shouldBe` Time (3 :% 1000000)
         it "7 days is 1 week" $
             convertUnit @WeekUnit (day 7) `shouldBe` 1
         it "2 fornights is 28 days" $
             convertUnit @DayUnit (fortnight 2) `shouldBe` 28
+    describe "Read Time Test" $ do
+        it "parses '42s' as 42 seconds" $
+            read @Second "42s" `shouldBe` sec 42
```

Comment:
Well, you don't need to specify `sec` and `hour` and others here because you already specified `@Second` in type of `read`. So, I guess you can either write `read @Second "42s" shouldBe 42` or `read "42s" shouldBe sec 42`. Maybe it's better to use `@`-style for consistency.